.github/workflows/getEnvironmentVariables.js

```diff
@@ -143,3 +196,63 @@ async function getTheNewestVersion() {
   );
   return response.data["tag_name"];
 }
+
+async function getServices(token) {
+  // us-east-1
+  const response = await fetch(
+    `https://staging-cloud.saleor.io/api/regions/us-east-1/services`,
+    {
+      method: "GET",
+      headers: {
+        Authorization: `Token ${token}`,
+        Accept: "application/json",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+    },
+  );
+  const responseInJson = await response.json();
+  return responseInJson;
+}
+
+async function getServiceWithVersionOneBeforeVersion(version, token) {
+  const services = await getServices(token);
+  const sandboxServices = services.filter(
+    element => element.service_type === "SANDBOX",
+  );
+  const sortedServices = sortServicesByVersion(sandboxServices);
+  const currentService = sortedServices.find(service => {
+    return service.name === `saleor-staging-v${getFormattedVersion(version)}`;
+  });
+  const serviceBeforeCurrent =
+    sortedServices[sortedServices.indexOf(currentService) + 1];
+  console.log(`Selected ${serviceBeforeCurrent.name} service`);
+  return serviceBeforeCurrent.name;
+}
+
+function sortServicesByVersion(services) {
+  // This function is used to sort environments by their version
+  // It returns sorted list of services in descending order
+
+  return services.sort(function(a, b) {
+    //Convert version from string to array eg. from "3.5.7" to [3, 5, 7]
+    // Where 3 is main version, 5 is major version and 7 is patch version
+
+    const versionASplittedToArray = a.version.split(/\D/);
+    const versionBSplittedToArray = b.version.split(/\D/);
+    const mainVersionNumberA = versionASplittedToArray[0];
+    const mainVersionNumberB = versionBSplittedToArray[0];
+    const majorVersionNumberA = versionASplittedToArray[1];
+    const majorVersionNumberB = versionBSplittedToArray[1];
+    const patchVersionNumberA = versionASplittedToArray[2];
+    const patchVersionNumberB = versionBSplittedToArray[2];
+
+    //Compare two versions
+    if (mainVersionNumberA !== mainVersionNumberB) {
+      return mainVersionNumberB - mainVersionNumberA;
+    } else if (majorVersionNumberA !== majorVersionNumberB) {
+      return majorVersionNumberB - majorVersionNumberA;
+    } else if (patchVersionNumberA !== patchVersionNumberB) {
+      return patchVersionNumberB - patchVersionNumberA;
+    } else return 0;
+  });
```

Comment:
Can be simplified by doing this:
```js
// <major>.<minor>.<patch>
a = version1_str.split(".", 3)
b = version2_str.split(".", 3)
cmp = (a > b) - (a < b)
```

`cmd` will contains one of: -1 (a is less than b), 0 (a is equal to b), or 1 (a is greater than b)

```suggestion
  return services.sort((svc1, svc2) => {
		var a = svc1.version.split(".", 3);
		var b = svc2.version.split(".", 3);
		return (a > b) - (a < b);
	}
);
```

Reply:
This solution won't always work. It will compare versions as a string, not as a number. So for eg. 
var version1_str = "3.2.2"
var version2_str = "3.10.4"
version1_str > version2_str, because 2 > 1
![Uploading String.prototype.split() - JavaScript | MDN 🔊 2023-01-25 13-56-27.png…]()

Reply:
I would assume in that case you can convert the strings into integers